BlockServer/core/active_config_holder.py

```diff
@@ -216,33 +225,52 @@ def iocs_changed(self):
         Returns:
             set, set, set : IOCs to start, IOCs to restart, IOCs to stop.
         """
-        # Look for modified IOCs
-        iocs_to_start, iocs_to_restart = _compare_ioc_properties(self._cached_config, self._config)
-        iocs_to_stop = set()
-
-        # Look for any new/changed components
-        for name, component in self._components.items():
-            if name in self._cached_components:
-                _start, _restart = _compare_ioc_properties(
-                    self._cached_components[name], self._components[name])
-
-                iocs_to_start |= _start
-                iocs_to_restart |= _restart
-            else:
-                for ioc_name in component.iocs.keys():
-                    iocs_to_start.add(ioc_name)
-
-        # Look for removed IOCs
-        for ioc_name in set(self._cached_config.iocs.keys()).difference(self._config.iocs.keys()):
-            iocs_to_stop.add(ioc_name)
-
-        # Look for any removed components
-        for name, component in self._cached_components.items():
-            if name not in self._components:
-                for ioc_name in component.iocs.keys():
-                    iocs_to_stop.add(ioc_name)
-
-        return iocs_to_start, iocs_to_restart, iocs_to_stop
+        def _get_config_iocs(config, components):
+            iocs = {}
+            for ioc_name, ioc in config.iocs.items():
+                iocs[ioc_name] = ioc
+
+            for name, component in components.items():
+                for ioc_name, ioc in component.iocs.items():
+                    iocs[ioc_name] = ioc
+            return iocs
+
```

Comment:
Just a comment on something to check - this loads iocs from the config and then from the components, so if an ioc is defined in both the config and a components then the component will win. Is the same logic applied elsewhere in the blockserver?

Reply:
It's not possible to define in both places through the GUI, and configchecker will also throw errors about this as a double-check.